var/spack/repos/builtin/packages/wrf-hydro/package.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import shutil
+import tempfile
+
+import llnl.util.tty as tty
+
+from spack import *
+
+
+class WrfHydro(Package):
+    """WRF-Hydro is a community modeling system and framework
+    for hydrologic modeling and model coupling.
+    """
+
+    homepage = "https://ral.ucar.edu/projects/wrf_hydro/overview"
+    git = "https://github.com/NCAR/wrf_hydro_nwm_public.git"
+    url = "https://github.com/NCAR/wrf_hydro_nwm_public/archive/refs/tags/v5.2.0.tar.gz"
+
+    maintainers = ["katrinaebennett", "ryanlcrumley", "daniellivingston"]
+
+    version("master", branch="master")
+    version(
+        "5.2.0",
+        sha256="7a3c95d52e1ef5681cc4c77f2c6a080dea6fd27c0e8b7ad30e205238062d9e87",
+    )
+    # version('5.2.0-rc3', sha256='85fd5bd2fc51a08cd8677c80718f4ee76bbd1eefb39db867e318cc4dd4ed11e0')
+    version(
+        "5.1.2",
+        sha256="203043916c94c597dd4204033715d0b2dc7907e2168cbe3dfef3cd9eef950eb7",
+    )
+    version(
+        "5.0.0",
+        sha256="b2b25cee73c032d3cee5a718a692c7585eb4d6311027c4352cad2eecf07384b7",
+    )
+
+    # WRF-Hydro supports multiple build configs, ostensibly
+    # set as flags in trunk/NDHMS/template/setEnvar.sh. Override here.
+    variant("debug", default=False, description="Builds in debug mode")
+    variant(
+        "hydro_d",
+        default=False,
+        description="Enable enhanced diagnostic output for debugging",
+    )
+    variant(
+        "spatial_soil",
+        default=True,
+        description="Enable spatially distributed parameters for NoahMP",
+    )
+    variant(
+        "wrf_hydro_rapid",
+        default=False,
+        description="Enable RAPID model",
+    )
+    variant(
+        "wrfio_ncd_large_file_support",
+        default=True,
+        description="Allow netCDF I/O of files larger than 2 GiB",
+    )
+    variant(
+        "ncep_wcoss",
+        default=False,
+        description="WCOSS file units",
+    )
+    variant(
+        "wrf_hydro_nudging",
+        default=False,
+        description="Enable streamflow nudging",
+    )
+
+    depends_on("pkgconfig", type=("build"))
+    depends_on("libtirpc")
+
+    # WRF-Hydro dependencies
+    depends_on("autoconf")
+    depends_on("automake")
+    depends_on("hdf5+fortran+hl+mpi")
+    depends_on("m4", type="build")
+    depends_on("libtool", type="build")
+    depends_on("mpi")
+    depends_on("netcdf-c@4.4.1.1:")
+    depends_on("netcdf-fortran@4.4.4:")
+    depends_on("jasper")
+
+    def setup_environment(self, spack_env, run_env):
```

Comment:
`setup_environment` is an older method and typically it is best to split it into `setup_build_environment` and `setup_run_environment` (since you only update `spack_env` here, you should only need `setup_build_environment`). `spack edit nco` has an example of defining `setup_build_environment`.